### ISSUE TITLE
6.3.3 shows unexpected coverage on Python 3.11

### ISSUE THREAD

**Issue by Nothing4You** [coveragepy/coveragepy#1375]
**Describe the bug**
Updating from 6.3.2 to 6.3.3 adds unexpected coverage on 3.11 runs.

**To Reproduce**
How can we reproduce the problem?  Please *be specific*. Don't link to a failing CI job. Answer the questions below:
1. What version of Python are you using?
3.11.0b1 (main, May  8 2022, 15:20:32) [GCC 9.4.0]
1. What version of coverage.py shows the problem? The output of `coverage debug sys` is helpful.
<details>
<summary>coverage debug sys</summary>

```
-- sys -------------------------------------------------------
               coverage_version: 6.3.3
                coverage_module: /home/runner/.local/lib/python3.11/site-packages/coverage/__init__.py
                         tracer: -none-
                        CTracer: available
           plugins.file_tracers: -none-
            plugins.configurers: -none-
      plugins.context_switchers: -none-
              configs_attempted: .coveragerc
                   configs_read: /home/runner/work/aiomysql/aiomysql/.coveragerc
                    config_file: /home/runner/work/aiomysql/aiomysql/.coveragerc
                config_contents: b'[run]\nbranch = True\nsource = aiomysql, tests\nomit = site-packages, .tox\n'
                      data_file: -none-
                         python: 3.11.0b1 (main, May  8 2022, 15:20:32) [GCC 9.4.0]
                       platform: Linux-5.13.0-1022-azure-x86_64-with-glibc2.31
                 implementation: CPython
                     executable: /opt/hostedtoolcache/Python/3.11.0-beta.1/x64/bin/python
                   def_encoding: utf-8
                    fs_encoding: utf-8
                            pid: 2589
                            cwd: /home/runner/work/aiomysql/aiomysql
                           path: /home/runner/.local/bin
                                 /opt/hostedtoolcache/Python/3.11.0-beta.1/x64/lib/python311.zip
                                 /opt/hostedtoolcache/Python/3.11.0-beta.1/x64/lib/python3.11
                                 /opt/hostedtoolcache/Python/3.11.0-beta.1/x64/lib/python3.11/lib-dynload
                                 /home/runner/.local/lib/python3.11/site-packages
                                 /opt/hostedtoolcache/Python/3.11.0-beta.1/x64/lib/python3.11/site-packages
                    environment: HOME = /home/runner
                                 PY_COLORS = 1
                   command_line: /home/runner/.local/bin/coverage debug sys
                sqlite3_version: 2.6.0
         sqlite3_sqlite_version: 3.31.1
             sqlite3_temp_store: 0
        sqlite3_compile_options: COMPILER=gcc-9.4.0; ENABLE_COLUMN_METADATA; ENABLE_DBSTAT_VTAB
                                 ENABLE_FTS3; ENABLE_FTS3_PARENTHESIS; ENABLE_FTS3_TOKENIZER
                                 ENABLE_FTS4; ENABLE_FTS5; ENABLE_JSON1
                                 ENABLE_LOAD_EXTENSION; ENABLE_PREUPDATE_HOOK; ENABLE_RTREE
                                 ENABLE_SESSION; ENABLE_STMTVTAB; ENABLE_UNLOCK_NOTIFY
                                 ENABLE_UPDATE_DELETE_LIMIT; HAVE_ISNAN; LIKE_DOESNT_MATCH_BLOBS
                                 MAX_SCHEMA_RETRY=25; MAX_VARIABLE_NUMBER=250000; OMIT_LOOKASIDE
                                 SECURE_DELETE; SOUNDEX; TEMP_STORE=1
                                 THREADSAFE=1; USE_URI
```
</details>

3. What versions of what packages do you have installed? The output of `pip freeze` is helpful.
<details>
<summary>pip freeze</summary>

```
aiomysql @ file:///home/runner/work/aiomysql/aiomysql/dist/aiomysql-0.1.2.dev3%2Bg1778841-py3-none-any.whl
alabaster==0.7.12
asttokens==2.0.5
attrs==21.4.0
Babel==2.10.1
backcall==0.2.0
bleach==5.0.0
certifi==2021.10.8
cffi==1.15.0
charset-normalizer==2.0.12
commonmark==0.9.1
coverage==6.3.3
cryptography==37.0.2
decorator==5.1.1
docutils==0.17.1
executing==0.8.3
flake8==4.0.1
idna==3.3
imagesize==1.3.0
importlib-metadata==4.11.3
iniconfig==1.1.1
ipdb==0.13.9
ipython==8.3.0
jedi==0.18.1
jeepney==0.8.0
Jinja2==3.1.2
keyring==23.5.0
MarkupSafe==2.1.1
matplotlib-inline==0.1.3
mccabe==0.6.1
packaging==21.3
parso==0.8.3
pexpect==4.8.0
pickleshare==0.7.5
pkginfo==1.8.2
pluggy==1.0.0
prompt-toolkit==3.0.29
ptyprocess==0.7.0
pure-eval==0.2.2
py==1.11.0
pycodestyle==2.8.0
pycparser==2.21
pyflakes==2.4.0
Pygments==2.12.0
PyMySQL==1.0.2
pyparsing==3.0.9
pytest==7.1.2
pytest-cov==3.0.0
pytest-sugar==0.9.4
pytz==2022.1
readme-renderer==35.0
requests==2.27.1
requests-toolbelt==0.9.1
rfc3986==2.0.0
rich==12.4.1
SecretStorage==3.3.2
six==1.16.0
snowballstemmer==2.2.0
Sphinx==4.5.0
sphinxcontrib-applehelp==1.0.2
sphinxcontrib-asyncio==0.3.0
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==2.0.0
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.5
SQLAlchemy==1.3.24
stack-data==0.2.0
termcolor==1.1.0
toml==0.10.2
tomli==2.0.1
traitlets==5.2.0
twine==4.0.0
urllib3==1.26.9
wcwidth==0.2.5
webencodings==0.5.1
zipp==3.8.0
```
</details>

4. What code shows the problem?  Give us a specific commit of a specific repo that we can check out. If you've already worked around the problem, please provide a commit before that fix.
https://github.com/aio-libs/aiomysql/pull/787 - e.g. https://github.com/aio-libs/aiomysql/pull/787/commits/acb2c3ac77dd8f7883660859388922553233d0d9
1. What commands did you run?
`pytest --capture=no --verbosity 2 --cov-report term --cov-report xml --cov aiomysql --cov tests ./tests --mysql-unix-socket "unix-mysql5.7=/tmp/run-mysql-5.7/mysql.sock" --mysql-address "tcp-mysql5.7=127.0.0.1:3306"`

**Expected behavior**
Coverage should only be present for executed code paths.

**Additional context**
For debugging I've created https://github.com/Nothing4You/aiomysql/pull/46.
One specific 3.11 run is https://github.com/Nothing4You/aiomysql/runs/6436451809?check_suite_focus=true, there's also xml and html coverage reports available at https://github.com/Nothing4You/aiomysql/suites/6508136042/artifacts/241970473
3.10 run: https://github.com/Nothing4You/aiomysql/runs/6436451599?check_suite_focus=true and reports https://github.com/Nothing4You/aiomysql/suites/6508136042/artifacts/241970464.
You can find all coverage reports from the CI run at the bottom of https://github.com/Nothing4You/aiomysql/actions/runs/2324901478.

For example, in `tests/test_issues.py`, there's 2 unexpected newly covered lines.
In `test_issue_17`, line 177 has coverage, while the rest of the skipped test has no coverage, as expected.
In `test_issue_35`, line 218 has coverage, while the rest of the skipped test has no coverage, as expected.

There's a few more cases that are marked as newly covered in 3.11, if needed I can document the others as well.
This new coverage only happens on 3.11, the same run on 3.7-3.10 does not have this coverage change.

**Comment by nedbat** [comment]
I think this might be the result of a CPython bug that was fixed after 3.11b1: https://github.com/python/cpython/pull/92722

Do you have the possibility of building a more recent 3.11 and trying it?

**Comment by Nothing4You** [comment]
Thanks for investigating, that does indeed appear to be the issue.

Just installed `Python 3.11.0b1+ (heads/3.11:748c83d, May 22 2022, 20:45:41) [Clang 13.1.6 (clang-1316.0.21.2.5)] on darwin` on my local machine and I don't see coverage with `coverage==6.3.3`.
`Python 3.11.0b1 (main, May 15 2022, 00:48:53) [Clang 13.1.6 (clang-1316.0.21.2.5)] on darwin` however shows the unexpected coverage.
